sormas-backend/src/main/java/de/symeda/sormas/backend/user/UserService.java

```diff
@@ -268,7 +268,13 @@ public List<UserReference> getUserReferences(
 		}
 		if (CollectionUtils.isNotEmpty(districtUuids)) {
 			Join<User, District> districtJoin = userRoot.join(User.DISTRICT, JoinType.LEFT);
-			filter = CriteriaBuilderHelper.and(cb, filter, cb.in(districtJoin.get(AbstractDomainObject.UUID)).value(districtUuids));
+			Join<User, Region> userRegionJoin = userRoot.join(User.REGION, JoinType.LEFT);
+			Join<Region, District> districtRegionJoin = userRegionJoin.join(Region.DISTRICTS, JoinType.LEFT);
+
+			Predicate districtFilter = cb.or(cb.in(districtJoin.get(AbstractDomainObject.UUID)).value(districtUuids),
+					cb.in(districtRegionJoin.get(AbstractDomainObject.UUID)).value(districtUuids));
+
+			filter = CriteriaBuilderHelper.and(cb, filter, districtFilter);
```

Comment:
the update of the filter adds the users from all the districts in the region of the case which is incorrect; the "responsible surveillance officer" dropdown should contain only users from the case district or users that have the same region jurisdiction as the case